### DISCUSSION TITLE
How to let a bot send a telegram.Message type to a channel?

### DISCUSSION THREAD

**Discussion by Marocco2** [python-telegram-bot/python-telegram-bot#2776] [Q&A] [answered]
Hi all!
I wanted to create a bot to act like a mirror to a channel. I thought that if I get the telegram.Message of a channel post, modify some contents and then resend it, it would be a game changer.
Unfortunately, I didn't find a function that make PTB able to send an entire telegram.Message to Telegram, or I miss something?
Is there a way to send this type of message without deconstructing it every time for every case?

Thanks

**Comment by zeroone2numeral2** [comment]
If you don't need the message content to be changed, you can use [`Bot.copy_message()`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.bot.html#telegram.Bot.copy_message) or directly [`Message.copy()`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.message.html#telegram.Message.copy)

> Is there a way to send this type of message without deconstructing it every time for every case?

If you need to edit some of the properties, you have to deconstruct it and pass the correct properties to the function to send messages

Maybe you can hack around with the `Message.to_dict()` and `Message.de_json()` methods of the `Message` class

**Reply by Marocco2** [reply]
> if you don't need the message content to be changed, you can use [`Bot.copy_message()`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.bot.html#telegram.Bot.copy_message) or directly [`Message.copy()`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.message.html#telegram.Message.copy)

Unfortunately I need message content to be changed and before being sent to the mirror channel



> If you need to edit some of the properties, you have to deconstruct it and pass the correct properties to the function to send messages

In that case, I'll check if there is a smart way to do that, thanks

**Reply by Marocco2** [reply]
> Maybe you can hack around with the `Message.to_dict()` and `Message.de_json()` methord of the `Message` class

Interesting, I'll check!

**Comment by Marocco2** [comment]
Looking into `bot._message()` I found that endpoints are hard-coded for every type. I will need to create a smart algo to check the right endpoint for every message 😥

**Comment by Bibo-Joshi** [comment, marked as answer]
[`send_by_kwargs`](../../ptbcontrib/blob/master/ptbcontrib/send_by_kwargs#send_by_kwargs) can probably help with what @zeroone2numeral2 suggested, i.e. you could do something like

```python
kwargs = update.effective_message.to_dict()
kwargs['text'] += 'foo'
send_by_kwargs(context.bot, kwargs)
```

**Reply by Marocco2** [reply]
Thank you! Seems right the perfect solution for my problem!
